pyproject.toml

```diff
@@ -127,6 +127,10 @@ module = [
 disallow_untyped_defs = true
 module = ["xarray.core.rolling_exp"]
 
+[[tool.mypy.overrides]]
+strict = true
```

Comment:
Super — should we put this above the `untyped_defs` and direct folks to aim for `strict`?

FYI here is the list — https://mypy.readthedocs.io/en/stable/existing_code.html#introduce-stricter-options

Reply:
Do you understand why the mypy ci is checking all files now, @max-sixty ?

Reply:
Erm... 

I do remember that `strict` wasn't previously possible to enable for a subset of modules, but then I thought that was changed...

One alternative is to try _some_ of the stricter options. I do think that the ROI on those is vastly different between options, such that expanding the modules we put into full `strict` might be difficult... (https://github.com/python/mypy/issues/7767)

FYI I put in #8242

Reply:
FYI this is sort of error with strict that might feel uphill...

```
xarray/namedarray/core.py:427: error: Call to untyped function "maybe_promote" in typed context  [no-untyped-call]
```

Reply:
`dtypes.py` will be vendored for namedarray so it won't be that scary to type the function then I think.